strawberryfields/gbs/data.py

```diff
@@ -0,0 +1,199 @@
+        matrix should be provided as a ``.npy`` binary. For ``_data_filename = "example"``,
+        the corresponding samples should be stored as ``./data/example.npz`` and the adjacency
+        matrix as ``./data/example_A.npy``."""
+        pass
+
+    def __init__(self):
+        """Instantiating ``Dataset`` causes the ``scipy.sparse.csr_matrix`` of samples and
+        ``numpy.ndarray`` adjacency matrix to be loaded."""
+        self._data = scipy.sparse.load_npz(DATA_PATH + self._data_filename + ".npz")
+        self.adj = np.load(DATA_PATH + self._data_filename + "_A.npy")
+        self.n_samples, self.modes = self._data.shape
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if self._count < self.n_samples:
+            self._count += 1
+            return self.__getitem__(self._count - 1)
+        self._count = 0
+        raise StopIteration
+
+    def _elem(self, i):
+        """Access the i-th element of the sparse array and output as a list."""
+        return list(self._data[i].toarray()[0])
+
+    def __getitem__(self, key):
+        if isinstance(key, (slice, tuple)):
+            if isinstance(key, tuple):
+                key = slice(*key)
+            range_tuple = key.indices(self.n_samples)
+            return [self._elem(i) for i in range(*range_tuple)]
+
+        if key < 0:
+            key += self.n_samples
+
+        return self._elem(key)
+
+    def counts(self, axis: int = 1) -> list:
+        """Count number of photons or clicks.
+
+        Counts number of photons/clicks in each sample (``axis==1``) or number of photons/clicks
+        in each mode compounded over all samples (``axis==0``).
+
+        Args:
+            axis (int): axis to perform count
+
+        Returns:
+            list: counts from samples
+        """
+        return np.array(self._data.sum(axis)).flatten().tolist()
+
+    @property
+    @abstractmethod
+    def n_mean(self) -> float:
+        """float: mean number of photons in the GBS device"""
+        pass
+
+    @property
+    @abstractmethod
+    def n_max(self) -> float:
+        """float: maximum number of photons allowed in any sample. This number is set to limit
+        the computation time, any sample being simulated that exceeds ``n_max`` will be ignored
+        and cause calculation to skip to the next sample."""
+        pass
+
+    @property
+    @abstractmethod
+    def threshold(self) -> bool:
+        """bool: flag to indicate whether samples are generated with threshold detection (i.e.,
+        detectors of zero or some photons) or with photon number resolving detectors."""
+        pass
+
+
+class Planted(Dataset):
+    """Dataset of samples generated from GBS with an embedded 30-node graph containing a dense
+    10-node subgraph planted inside :cite:`arrazola2018using`.
+
+    Attributes:
+        n_mean = 6
+        n_max = 14
+        threshold = True
+        n_samples = 50000
+        modes = 30

```

Comment:
I feel this should be a bullet point list with backticks

Reply:
I guess Josh's comment is that it would look better with backticks? Would be nice to compare both if it's not too much trouble

Reply:
![image](https://user-images.githubusercontent.com/49409390/65920399-e6f1d800-e3ac-11e9-9a3d-3a65cd04e657.png)

^ Although maybe I need to put the backticks in some other configuration?

Reply:
Ah, I was meaning the formatting in the actual code docstring, not the rendered page. But I didn't know you could write it like that and it would render! In which case I'm happy with it as is. Learnt something knew today 💪